platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -245,291 +518,180 @@ message RoundRosterPair {
-  AddressBook from `config.txt` will be removed or refactored and moved into the Services codebase.
+#### Roster Store APIs
 
-The current network Transplant procedure is manual. DevOps is given a State and `config.txt` file on disk. This is
-then followed by a config-only upgrade to adopt the `config.txt` file using that state. This will change to the
-use of a Genesis Roster or Override Roster. The new startup sequence will be determined by the presence of a Genesis
-Roster, State, or Override Roster as described in the following sections.
-
-### The new Startup Behaviour
-
-![](TSS%20Roster%20Lifecycle-Proposed%20Startup%20Behavior.drawio.svg)
-
-#### Genesis Roster
-
-A `Genesis Roster` is an optional Roster that DevOps may provide for the explicit purpose of starting a Genesis Network
-Process. It is essential to distinguish the Genesis Roster from the Candidate, Active, or Override rosters. The
-Genesis Roster is a special roster used for the sole purpose of bootstrapping a brand-new Genesis network; one in
-which there is no existing State, and the round number begins at 1. However, its structure is exactly the same as
-the Active or Candidate Rosters.
-
-The equivalent of the Genesis Roster in the current DevOps flow is the `config.txt` file. DevOps and Services may
-choose to create the Genesis Roster from the `config.txt` file or some other mechanism. The trigger for this
-workflow will be the presence of a Genesis Roster at start-up with no State on disk.
-
-A new method will be added to the `PlatformBuilder` that will be used by the App to set the Genesis Roster.
-In most cases, there will be no need to call the method on `PlatformBuilder` to set the Genesis Roster except in the
-events we need to bootstrap a Genesis Network.
+##### ReadableRosterStore
 
 ```java
-//in PlatformBuilder
-
-/*
-  Set the Genesis Roster for the network.
-  <p>
-  This method is used to set the optional Genesis Roster for the network.
-  The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network.
-  The Genesis Roster, if set, is immutable and will be used to start a Genesis network Process or Network Transplant process.
-  <p>
-  @param genesisRoster the Genesis Roster for the network.
+/**
+ * Read-only implementation for accessing rosters states.
  */
-PlatformBuilder withGenesisRoster(@NonNull final Roster genesisRoster);
-```
-
-The logic to determine whether to invoke this method could be determined by some mechanism Services deems appropriate.
-
-#### Override Roster
+public interface ReadableRosterStore {
+    /**
+     * Gets the candidate roster if found in state or null otherwise.
+     * Note that state commits are buffered,
+     * so it is possible that a recently stored candidate roster is still in the batched changes and not yet committed.
+     * Therefore, callers of this API must bear in mind that an immediate call after storing a candidate roster may return null.
```

Comment:
I still do not understand why we have to insist on this behavior. It breaks with the commonly expected behavior of getters and setters and differs from all other stores that exist in consensus node.

Reply:
I did not write that text.   I would be happy to update it to whatever it needs to be.   I think I copied it from the code base, which has @derektriley as the committer.